enhancements/authentication/unsupported-direct-use-of-oidc.md

```diff
@@ -0,0 +1,368 @@
+---
+title: direct-use-of-oidc
+authors:
+  - "@stlaz"
+reviewers:
+  - "@s-urbaniak"
+  - "@liouk"
+  - "@ibihim"
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+  - "@s-urbaniak"
+api-approvers:
+  - "@deads2k"
+  - "@tkashem"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: []
+see-also:
+  - [oc Authorization Grant login](./improved-login-workflow.md)
+  - [Configuring OIDC directly - study](./direct-oidc-study/study-oidc-in-openshift.md)
+replaces: []
+superseded-by: []
+---
+
+# Direct use of OIDC as an identity provider
+
+## Summary
+
+When users authenticate to OpenShift, the authentication goes through the `oauth-server`
+which serves as middle-man actor that unifies access to multiple kinds of
+identity providers, such as LDAP, OIDC providers, providers returning HTTP Basic
+Authentication challenges, and similar.
+
+The above differs from vanilla Kubernetes where authentication is dealt directly
+by the `kube-apiserver`. This has limitations especially when it comes to being
+able to login to different kinds of identity providers in a uniform manner.
+On the other hand, `kubectl`, the binary for accessing Kubernetes APIs, has evolved
+through the years to simplify the login process to Kubernetes clusters, and communities
+built useful toolings around the binary.
```

Comment:
We cannot just parse an OIDC token and issue a token of our own because there is no OAuth2 spec that would allow this very exchange.

If we really wanted, we could try to implement the OAuth2 Token Exchange based on `id_tokens` but then we might struggle with how supported this would be on the OIDC side.

Reply:
I thought of it as a kind of authentication as we do with htpasswd. Non-OAuth2 conformant. We would look up the public key from a config, verify the signature and the authenticate.